src/app/clients/AddParcelsButton.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import React from "react";
+import { ClientsTableRow } from "@/app/clients/getClientsTableData";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import PopUpButton from "@/components/Buttons/PopUpButton";
+import LinkButton from "@/components/Buttons/LinkButton";
+import Button from "@mui/material/Button";
+import TableSurface from "@/components/Tables/TableSurface";
+import Table, { Datum, TableHeaders } from "@/components/Tables/Table";
+import styled from "styled-components";
+import { Schema } from "@/supabase";
+
+interface Props {
+    data: ClientsTableRow[];
+}
+
+interface ClientsListRow extends Datum {
+    primaryKey: string;
+    fullName: Schema["clients"]["full_name"];
+    familyCategory: string;
+    addressPostcode: Schema["clients"]["address_postcode"];
+}
+
+const headers: TableHeaders = [
+    ["fullName", "Name"],
+    ["familyCategory", "Family"],
+    ["addressPostcode", "Postcode"],
+];
+
+const PopUp = styled.div`
+    display: flex;
+    flex-direction: column;
+    text-align: center;
+    padding: 2rem;
+`;
+
+const styleOptions = {
+    fullName: {
+        minWidth: "8rem",
+    },
+    familyCategory: {
+        hide: 550,
+    },
+    addressPostcode: {
+        hide: 800,
+    },
+};
+
+const showClients = (data: ClientsTableRow[]): ClientsListRow[] => {
+    const primaryKeys: string[] = [];
+    const clientsData: ClientsListRow[] = [];
+
+    for (const datum of data) {
+        if (!primaryKeys.includes(datum.primaryKey)) {
+            const { primaryKey, fullName, familyCategory, addressPostcode } = datum;
+            clientsData.push({ primaryKey, fullName, familyCategory, addressPostcode });
```

Comment:
can this not be `clients.Data.push(datum)`

Reply:
no, because I only want some of the propersties of datum and not the rest